app/controllers/foreman_wreckingball/hosts_controller.rb

```diff
@@ -52,22 +52,21 @@ def status_dashboard
 
     # ajax method
     def status_hosts
-      statuses_map = {
-        vmware_tools_status_object: ForemanWreckingball::ToolsStatus,
-        vmware_operatingsystem_status_object: ForemanWreckingball::OperatingsystemStatus,
-        vmware_cpu_hot_add_status_object: ForemanWreckingball::CpuHotAddStatus,
-        vmware_spectre_v2_status_object: ForemanWreckingball::SpectreV2Status,
-        vmware_hardware_version_status_object: ForemanWreckingball::HardwareVersionStatus
-      }
-
-      @status = statuses_map[params[:status].to_sym]
+      @status = STATUSES_MAP[params.fetch(:status).to_sym]
+
+      total = Host.authorized(:view_hosts, Host)
+                  .joins(@status.host_association)
+                  .includes(@status.host_association, :vmware_facet, :environment)
+                  .count
+
       all_hosts = Host.authorized(:view_hosts, Host)
                       .joins(@status.host_association)
                       .includes(@status.host_association, :vmware_facet, :environment)
+                      .where('"host_status"."status" NOT IN (:status)', status: @status.global_ok_list)
                       .preload(:owner)
                       .order(:name)
-                      .reject { |h| h.send(@status.host_association).to_global == HostStatus::Global::OK }
-      @count = all_hosts.count
+
+      @count = total
```

Comment:
why you replaced `@count = all_hosts.count` with `@count = total`? I think `all_hosts.count` is correct number

![image](https://user-images.githubusercontent.com/37402924/48834067-20b18580-ed7d-11e8-8b29-01919924ef3d.png)

![image](https://user-images.githubusercontent.com/37402924/48833728-5144ef80-ed7c-11e8-9077-f4455ed5ac94.png)
there are 87 hosts in the table but as you can see, numbers are incorrect now,
should be `Showing 1 to 87 of 87 Items (of 87)`

Reply:
It was probably a matter of interpretation what `recordsFiltered` means. One issue I saw was, pagination still accounted for the total number of all hosts (including all healthy ones), with the effect of listing all pages (e.g. page 1 of 42), while only 20-30 hosts were listed. Clicking on to the page 2 for example resulted in an error message. Not sure if the user should be allowed to navigate between pages even though the results all fit on the first page.